### PR TITLE
The /components page

### PR DESCRIPTION
**Pull request by nottrobin**
The data might be a bit different. The search doesn't work the same way, because we're waiting for the `query` API parameter to be implemented.

Update: Now the `/components/{id}` endpoints work

QA
--

`./run`

- [ ] go to `/components` - check it works.
- [ ] click on some components, check the corresponding details page works fine
- [ ] Go to a hardware page with components, e.g. `/hardware/201711-25981`, check you see a "Component devices" table, and you can click through to `/components?query=201711-25981/` (although that URL won't work yet until the API query supports searching canonical_ids - @codersquid is working on it)

### CODE REVIEW

**Comment by jpmartinspt** [comment]
In the new version we seem to be missing the (1) annotation in some components. For example: Ellesmere [Radeon Pro WX 5100].

**Comment by nottrobin** [comment]
Wow you're quick. I'd fixed that but forgot to push. Pushing now.

**Review by jpmartinspt — DISMISSED** [review]

**Comment by nottrobin** [comment]
@jpmartinspt I expanded this PR to add a couple of other things. Would you mind reviewing it again?

File: webapp/app.py [hunk 1]
```diff
@@ -385,3 +388,68 @@ def soc_models():
...
+
+@app.route("/components/<id>")
+def component_details(id):
+    component = api.componentsummary(id)
+
+    all_machines = api.certifiedmodels(
+        canonical_id__in=component["machine_canonical_ids"]
+    )["objects"]
+
+    machines_by_id = {}
```
Comment (jpmartinspt) [inline webapp/app.py:441]: Is this because the API returns duplicates?

Comment (nottrobin) [reply webapp/app.py:441]: yep

**Review by jpmartinspt — APPROVED** [review]

**Pull request merged**